pkg/providers/vsphere/session/session_vm_update.go

```diff
@@ -798,27 +800,26 @@ func (s *Session) poweredOnVMReconfigure(
 		}
 	}
 
-	var refetchProps bool
+	refetchProps, err := doReconfigure(
+		logr.NewContext(
+			vmCtx,
+			vmCtx.Logger.WithName("poweredOnVMReconfigure"),
+		),
+		vmCtx.VM,
+		resVM.VcVM(),
+		*configSpec)
 
-	defaultConfigSpec := &vimtypes.VirtualMachineConfigSpec{}
-	if !apiEquality.Semantic.DeepEqual(configSpec, defaultConfigSpec) {
-		vmCtx.Logger.Info("PoweredOn Reconfigure", "configSpec", configSpec)
-		if _, err := resVM.Reconfigure(vmCtx, configSpec); err != nil {
-			vmCtx.Logger.Error(err, "powered on reconfigure failed")
-			return false, err
-		}
+	if err != nil {
+		return false, err
+	}
 
-		// Special case for CBT: in order for CBT change take effect for a powered on VM,
-		// a checkpoint save/restore is needed.  tracks the implementation of
-		// this FSR internally to vSphere.
-		if configSpec.ChangeTrackingEnabled != nil {
-			if err := s.invokeFsrVirtualMachine(vmCtx, resVM); err != nil {
-				vmCtx.Logger.Error(err, "Failed to invoke FSR for CBT update")
-				return false, err
-			}
+	// Special case for CBT: in order for CBT change take effect for a powered on VM,
+	// a checkpoint save/restore is needed.  tracks the implementation of
+	// this FSR internally to vSphere.
+	if refetchProps && configSpec.ChangeTrackingEnabled != nil {
```

Comment:
Why does this need to check `refetchProps`?

Reply:
Sorry, I know it's not super clear due to the variable name, but `refetchProps` is also `didReconfigureHappen`. Thus, we do not need to issue the FSR unless the reconfig actually occurred.

Reply:
I updated the comments to reflect this.